File: plugins/java-kit/testdata/good/src/main/java/demo/Ok.java
package demo;

import java.lang.System.Logger;
import java.lang.System.Logger.Level;
import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;

@org.jspecify.annotations.NullMarked
public class Ok {
  private static final Logger LOG = System.getLogger(Ok.class.getName());

  public void logOk(String message) {
    LOG.log(Level.INFO, message);
  }

  public ResultSet sqlOk(Connection conn, long id) throws Exception {
    PreparedStatement ps = conn.prepareStatement("SELECT name FROM users WHERE id = ?");
    ps.setLong(1, id);
    return ps.executeQuery();
  }

  /* Named boundary docs; intentional legacy uses java-rg-allow on the smell line. */
  public void documentedLegacy() {
    System.out.println("boot banner"); /* java-rg-allow: process bootstrap before logger binds */
  }
}

File: plugins/csharp-kit/testdata/good/src/Ok.cs
using System.Data;
using System.Threading;
using System.Threading.Tasks;
using Microsoft.Data.SqlClient;
using Microsoft.Extensions.Logging;

namespace Demo;

public class Ok
{
    private readonly ILogger<Ok> _log;

    public Ok(ILogger<Ok> log)
    {
        _log = log;
    }

    public void LogOk(string message)
    {
        _log.LogInformation("{Message}", message);
    }

    public async Task<string?> SqlOkAsync(SqlConnection conn, long id, CancellationToken ct)
    {
        await using var cmd = new SqlCommand("SELECT name FROM users WHERE id = @id", conn);
        cmd.Parameters.Add(new SqlParameter("@id", SqlDbType.BigInt) { Value = id });
        var result = await cmd.ExecuteScalarAsync(ct);
        return result as string;
    }

    /* Named boundary docs; intentional legacy uses csharp-rg-allow on the smell line. */
    public void DocumentedLegacy()
    {
        Console.WriteLine("boot banner"); // csharp-rg-allow: process bootstrap before logger binds
    }
}
